### ISSUE TITLE
Linking problem when trying an example in ubuntu 16.04 LTS i.e undefined reference problem.

### ISSUE THREAD

**Issue by saaelayaz** [memo/ofxMSATensorFlow#33]
Hi there memo,

So I followed up your instruction you mentioned in the [https://github.com/memo/ofxMSATensorFlow/wiki/Getting-started](url), when whenever I try to make any of the example, example-basic for instance I end up with compilation (link) error.

Screenshot is here,

![screenshot from 2017-11-08 16-31-35](https://user-images.githubusercontent.com/33484357/32546954-8286cffe-c4a2-11e7-83c7-78829d77ada6.png)
![screenshot from 2017-11-08 16-31-53](https://user-images.githubusercontent.com/33484357/32546964-868dd9c6-c4a2-11e7-80cd-33515dfc6c7a.png)
![screenshot from 2017-11-08 16-32-09](https://user-images.githubusercontent.com/33484357/32546970-8938bb3c-c4a2-11e7-90ab-089ae4759ba0.png)

I have been struggling with this for 3 days, tried various things. As I'm using your libtensorflow_cc.so file for tensorflow r1 I downloaded your addon for tensorflow r1 but that didn't work. It would be great if you could help me. 

Thanks pal for the addon BTW. I mean seriously, great work :)

**Comment by memo** [comment]
Hi, is there a reason you're using r1.0? I can't remember if there were any peculiarities with my r1.0 lib, so I'd recommend the current version which is r1.3 (actually they just released tf1.4! argh, will build that when I get a chance).  That lib is built for CUDA8 and cuDNN6.

The error looks like it can't find any of the symbols defined in the lib. Did you follow the step "Copy library and set search path". Copying the lib into the ofxMSATensorFlow/.../lib folder isn't enough, you also need to put it in a system libs path (and run sudo ldconfig).

**Comment by saaelayaz** [comment]
Hello @memo that's very kind of you that you replied,

For the second part, yeah I did that. Actually you wrote in the wikis that we have many options, so I preferred the script you wrote, that will copy the library from the actual folder put it in the ./lib path and write the conf file in the /etc/ld.so.config.d/ and then run "sudo ldconfig".

Now for the first part, actually I went for the r1.0 coz the master branch of your addon didn't work in the first place so I was compelled to try out the previous version to check if they work. Also the build for r1.3 is for CUDA and cuDNN6 and my machine don't have NVIDIA it's running on intel HD graphics. I'm using HP folio 9470 m.

So that's the case, please suggest something useful man, I'm dying to use your addon coz I have to complete my smart drone project that uses tensorflow.

**Comment by saaelayaz** [comment]
Sorry I forgot to mention that I did tried to compile the r1.4 of the tensorflow they just released on my laptop, following your steps "Rebuilding library from scratch", using bazel and their ./configure file. But, all the 3 times I tried my laptop crashed. It was just so painful, I didn't wanted to do that again.
*Seems like I need to purchase more powerful machine :)

Also, I don't think it's lib r1.0 problem, coz I tried lib 0.7 too. The errors were the same.

**Comment by memo** [comment]
ah, yes I realise I haven't done a non-GPU build since r1.0, sorry about that :P I'll try to do that for the next version.

I presume when you use the r1.0 lib, you also checkout the tf1.0 tag for the addon src?

I'm afraid without digging deeper I don't know what your problem is I'm afraid. 

(I have built (compiled) r1.4 lib actually, but there are issues with the headers I need to sort out. )

**Comment by saaelayaz** [comment]
@memo  Yeah absolutely, but when I use the sources for r1.0 lib then I use same version of addon i.e r1.0, but in that case there is another huge list of errors.

Okay pal, I'll wait for your r1.4 lib (Non-GPU please) but my humble request is please do that ASAP. In next couple of days or a week may be?

But I suspect that ultimately I'll be using your addon on raspberry pi 3 and then there would be somewhat new issues, I mean with lib.so what do you think? Or it would work just fine?

**Comment by saaelayaz** [comment]
After struggling about a week almost, I got the solution.

So here is what I was wrong at, I was using the addon version r1.3, but I was linking it with a tensorflow library r1.0 so it wasn't linking properly.

I tried to build tensorflow r1.3 myself but my laptop crashed, so what I did was to setup swaping memory for my system, it don't let your system crash.
***Links [https://help.ubuntu.com/community/SwapFaq](url) for how to configure swapping memory.
